### PR TITLE
Move POSTGRES backend to the backends/ directory

### LINKED ISSUE #2421: Move POSTGRES backend to the backends/ directory
**Move the sql/postgres/` backend to `backends/postgres`.**

The current Ibis directory structure is a bit confusing, since backends are mixed with core modules. For example, it's not obvious to know if `pandas/` is a backend, or core functionality. Many of the backends live in the root directory, and some are grouped, like in a directory `sql/`. This makes things even harder and difficult to maintain. Also, ibis uses attributes to access the backends (e.g. `ibis.spark.connect`), and `ibis.spark` in code seems to be the `ibis/spark/` module, but it's not, making things confusing, and even creating problems with pytest.

We're moving backends to a standardized structure, in the `backends/` directory, one backend at a time (each backend in a separate pull request). We started by the OmniSciDB backend, and its PR can be a reference for the changes that need to be made here: #2392

The exact tasks to perform are:
- Move the backend from its original location to `backends/` (if the backend is in a subdirectory, like `sql/`, ignore the subdirectory and just have the backend inside `backends/`, so `sql/sqlite/` -> `backends/sqlite`)
- Move the content of `backends/<your-backend>/api.py` to `backends/<your-backend>/__init__.py`. Backends are using `api.py` for historical reasons, but the public backend API should be in the standard `__init__.py` file of the module.
- Update `ibis/__init__.py` so the backend is imported from the new module (`from ibis.<your-backend> import api` -> `from ibis import <your-backend>`)
- Look for places in the code where your backend is being imported, and update the imports. For imports inside the same backend, please use relative imports (e.g. `import ibis.<your-backend>.whatever` -> `import .whatever`). See the reference PR for examples.
- Update the documentation (in `doc/source/backends/<your-backend>.rst`) so the API is generated from the right location.

After performing the changes, the CI should be green, except for the packaging build (which requires changes in the conda-forge feedstock). You can ignore that error, those changes will be performed later for all moved backends.

### PR DESCRIPTION
**Pull request by jcorrius**
Closes #2421

### CODE REVIEW

File: ibis/__init__.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 
 with suppress(ImportError):
     # pip install ibis-framework[postgres]
-    import ibis.sql.postgres.api as postgres  # noqa: F401
+    import ibis.backends.postgres.api as postgres  # noqa: F401
```
Comment (datapythonista) [inline ibis/__init__.py:40]:
```suggestion
    from ibis.backends import postgres  # noqa: F401
```

**Review by datapythonista — COMMENTED** [review]
lgtm, just one small thing

File: ibis/__init__.py (hunk 1 above)
Comment (jcorrius) [reply ibis/__init__.py]: yes, it's obvious. sorry for that.

File: ibis/backends/postgres/__init__.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
-from ibis.sql.alchemy import to_sqlalchemy
-from ibis.sql.postgres.client import PostgreSQLClient
-from ibis.sql.postgres.compiler import (  # noqa: F401, E501
+"""PostgreSQL backend."""
+from ibis.backends.postgres.client import PostgreSQLClient
+from ibis.backends.postgres.compiler import (  # noqa: F401, E501
```
Comment (datapythonista) [inline ibis/backends/postgres/__init__.py:3]:
```suggestion
from .client import PostgreSQLClient
from .compiler import (  # noqa: F401, E501
```

**Review by datapythonista — COMMENTED** [review]
Looks good, the CI changes are unrelated. Just couple of imports that would be better if they are relative (you can make the change from the GitHub UI).

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @jcorrius 

The CI should be fixed soon, I'll ping you to merge master, and we should be ready to merge.

**Comment by datapythonista** [comment]
@jcorrius CI problems should be fixed now. You'll have to merge master into your branch, and also change the module in https://github.com/ibis-project/ibis/blob/master/ci/recipe/meta.yaml#L64 to get the CI passing (`ibis.postgres` -> `ibis.backends.postgres`).

**Comment by datapythonista** [comment]
Changes look good, but some tests seem to be broken. Not sure why, it's complaining that in `import ibis; ibis.postgres`, the attribute `postgres` is not found (even it it looks correctly set in `ibis/__init__.py`): https://github.com/ibis-project/ibis/pull/2432/checks?check_run_id=1210321162#step:5:7330

Can you have a look?

File: ibis/__init__.py [hunk 3]
```diff
@@ -37,7 +37,7 @@
 
 with suppress(ImportError):
```
Comment (datapythonista) [inline ibis/__init__.py:38]: Can you try to temporary remove this with? It may be hiding the actual error.

Comment (jcorrius) [reply ibis/__init__.py:38]: Done. New commit pushed to the PR.

**Comment by datapythonista** [comment]
Seems like the `Server Error` problems in azure are fixed now, all green.

@jreback this should be ready. Let me know if you want to review the PRs moving the backends, or you want me to merge them directly, they should all be the same.

**Comment by jreback** [comment]
let me look at one

**Comment by jreback** [comment]
lgtm (there is a conflict which i think i just caused by merging isort). if you'd fix up and merge.

**Comment by datapythonista** [comment]
Thanks @jcorrius for helping with this, very useful contribution.

**Pull request merged**